zadanie4/app.rb

```diff
@@ -0,0 +1,49 @@
+require 'bundler'
+require 'digest'
+require 'erb'
+require 'fast_secure_compare/fast_secure_compare'
+
+Bundler.require(:default)
+
+require 'sinatra/reloader'
+enable :sessions
+set :sessions, expire_after: 2_592_000
+set :session_store, Rack::Session::Pool
+
+def logged_in?
+  !session[:id].nil?
+end
+
+get '/' do
+  if logged_in?
+    redirect '/logged'
+  else
+    erb :home
+  end
+end
+
+post '/login' do
+  password = params['password']
+  stored_pw = File.read('password.txt')
```

Comment:
Jeśli hasło z pliku się nie zmienia to efektywniej jest je odczytać tylko raz i potem przetrzymywać tę informację w stałej w pamięci do której dostęp jest szybszy.